python/pycylon/pycylon/data/table.pyx

```diff
@@ -1260,10 +1310,30 @@ cdef class Table:
         tb_index = self.index
         py_arrow_table = self.to_arrow().combine_chunks()
         if isinstance(key, slice):
-            new_tb = self.from_arrow(self.context, py_arrow_table.slice(key.start, key.stop))
-            new_index = tb_index.values[key.start: key.stop].tolist()
-            new_tb.set_index(new_index)
-            return new_tb
+            if key.start != None:
+                new_tb = self.from_arrow(self.context, py_arrow_table.slice(key.start, key.stop))
+                new_index = tb_index.values[key.start: key.stop].tolist()
+                new_tb.set_index(new_index)
+                return new_tb
+            elif key.start == None and key.stop > 0:
+                return self._get_head_response(0, key.stop)
+            elif key.start == None and key.stop < 0:
+                    return self._get_tail_response(0, key.stop*(-1))
+            else:
+                raise ValueError(f"Unsupported Key Type in __getitem__ {type(key)}")
```

Comment:
Thanks for adding this. This was something missing in the initial implementation.

Reply:
@vibhatha 
This is a wrong pull request. Niranda is supposed to close this.

Please review this one : https://github.com/cylondata/cylon/pull/619

Reply:
Yep. My bad! Sorry @arupcsedu and @vibhatha.